go/adbc/driver/databricks/record_builder.go

```diff
@@ -0,0 +1,338 @@
+package databricks
+
+import (
+	"fmt"
+	"time"
+
+	"github.com/apache/arrow-go/v18/arrow"
+	"github.com/apache/arrow-go/v18/arrow/array"
+	"github.com/apache/arrow-go/v18/arrow/memory"
+)
+
+// ColumnBuilder is an interface for building Arrow arrays
+type ColumnBuilder interface {
+	Append(value interface{}) error
+	Builder() array.Builder
+}
+
+// StringColumnBuilder handles string values
+type StringColumnBuilder struct {
+	builder *array.StringBuilder
+}
+
+func NewStringColumnBuilder(mem memory.Allocator) *StringColumnBuilder {
+	return &StringColumnBuilder{
+		builder: array.NewStringBuilder(mem),
+	}
+}
+
+func (b *StringColumnBuilder) Append(value interface{}) error {
+	if value == nil {
+		b.builder.AppendNull()
+		return nil
+	}
+	if str, ok := value.(string); ok {
+		b.builder.Append(str)
+	} else {
+		b.builder.Append(fmt.Sprintf("%v", value))
+	}
+	return nil
+}
+
+func (b *StringColumnBuilder) Builder() array.Builder {
+	return b.builder
+}
+
+// Int32ColumnBuilder handles int32 values
+type Int32ColumnBuilder struct {
+	builder *array.Int32Builder
+}
+
+func NewInt32ColumnBuilder(mem memory.Allocator) *Int32ColumnBuilder {
+	return &Int32ColumnBuilder{
+		builder: array.NewInt32Builder(mem),
+	}
+}
+
+func (b *Int32ColumnBuilder) Append(value interface{}) error {
+	if value == nil {
+		b.builder.AppendNull()
+		return nil
+	}
+	switch v := value.(type) {
+	case int:
+		b.builder.Append(int32(v))
+	case int32:
+		b.builder.Append(v)
+	default:
+		return fmt.Errorf("invalid type for Int32 column: %T", value)
+	}
+	return nil
+}
+
+func (b *Int32ColumnBuilder) Builder() array.Builder {
+	return b.builder
+}
+
+// Int64ColumnBuilder handles int64 values
+type Int64ColumnBuilder struct {
+	builder *array.Int64Builder
+}
+
+func NewInt64ColumnBuilder(mem memory.Allocator) *Int64ColumnBuilder {
+	return &Int64ColumnBuilder{
+		builder: array.NewInt64Builder(mem),
+	}
+}
+
+func (b *Int64ColumnBuilder) Append(value interface{}) error {
+	if value == nil {
+		b.builder.AppendNull()
+		return nil
+	}
+	switch v := value.(type) {
+	case int64:
+		b.builder.Append(v)
+	case int:
+		b.builder.Append(int64(v))
+	default:
+		return fmt.Errorf("invalid type for Int64 column: %T", value)
+	}
+	return nil
+}
+
+func (b *Int64ColumnBuilder) Builder() array.Builder {
+	return b.builder
+}
+
+// Float32ColumnBuilder handles float32 values
+type Float32ColumnBuilder struct {
+	builder *array.Float32Builder
+}
+
+func NewFloat32ColumnBuilder(mem memory.Allocator) *Float32ColumnBuilder {
+	return &Float32ColumnBuilder{
+		builder: array.NewFloat32Builder(mem),
+	}
+}
+
+func (b *Float32ColumnBuilder) Append(value interface{}) error {
+	if value == nil {
+		b.builder.AppendNull()
+		return nil
+	}
+	switch v := value.(type) {
+	case float32:
+		b.builder.Append(v)
+	case float64:
+		b.builder.Append(float32(v))
```

Comment:
Does this case actually happen?

Reply:
yes the relationship between the go datatype of the object databricks returns is only loosely related to the schema they specify

Reply:
does `float32(v)` panic if v can't fit in a float32 (good) or it truncates the value somehow leading to data loss (really bad)?